src/dmd/ctfeexpr.d

```diff
@@ -247,10 +247,26 @@ extern (C++) final class CTFEExp : Expression
     }
 }
 
+/*
+ * This class is used to pass additional information
+ * about ctfe errors. For now it is used solely to
+ * signal a missing function body error but it can be
+ * extended to support more errors in the future, if needed,
+ * by adding a field which will represent the error.
+ */
+extern (C++) class CTFEErrorExtraInfoExp : Expression
```

Comment:
(1) I don't think creating a new expression is necessary....

Reply:
I think it's better to have a class that can be easily extended for other ctfe error types that need extra information.